core/src/test/java/ai/timefold/solver/core/api/solver/SolutionManagerTest.java

```diff
@@ -459,14 +509,70 @@ void recommendFitWithUnassigned(SolutionManagerSource SolutionManagerSource) {
 
     @ParameterizedTest
     @EnumSource(SolutionManagerSource.class)
-    void recommendFitUninitializedSolutionWithUnassigned(SolutionManagerSource SolutionManagerSource) {
+    void recommendAssignmentWithAllAssigned(SolutionManagerSource SolutionManagerSource) {
+        int valueSize = 3;
+        var solution = TestdataAllowsUnassignedSolution.generateSolution(valueSize, 3);
+        var evaluatedEntity = solution.getEntityList().get(2);
+
+        // At this point, entity 0 is unassigned.
+        // Entity 1 is assigned to value #1.
+        // But only entity2 should be processed for recommendations.
+        var solutionManager = SolutionManagerSource.createSolutionManager(SOLVER_FACTORY_UNASSIGNED);
+        assertThat(solutionManager).isNotNull();
+        var recommendationList =
+                solutionManager.recommendAssignment(solution, evaluatedEntity, TestdataAllowsUnassignedEntity::getValue);
+
+        // Three values means there need to be four recommendations, one extra for unassigned.
+        assertThat(recommendationList).hasSize(valueSize + 1);
```

Comment:
How will this unassigned recommendation work if we already unassign it at the beginning of the `AssignmentProcessor` logic?

Reply:
The processor only unassigns the value if already assigned.
It has to, because otherwise CH wouldn't do anything. (= All values already assigned.)

This is an implementation detail. There were two options to choose from for implementing this feature:

- Either articially unassign a value to make CH work as usual.
- Or refactor CH to be able to "ignore" that all entities are already assign, and try to assign one element anyway.

I decided that the former is not only easier, but also more natural.
It has no impact on the solution on the outside, as all of this only ever happens on the internal, cloned solution.

The test proves that the code works as intended.